tinygrad/helpers.py

```diff
@@ -41,7 +41,10 @@ def fully_flatten(l):
   return [l]
 def fromimport(mod, frm): return getattr(__import__(mod, fromlist=[frm]), frm)
 def strip_parens(fst:str): return fst[1:-1] if fst[0] == '(' and fst[-1] == ')' and fst[1:-1].find('(') <= fst[1:-1].find(')') else fst
-def ceildiv(num, amt): return int(ret) if isinstance((ret:=-(num//-amt)), float) else ret
+def ceildiv(num, amt):
+  if hasattr(num, 'op'): return -(num.floor_divide(-amt))  # XXX: isinstance(num, UOp) is circular import
+  if hasattr(amt, 'op'): return -(-amt.floor_divide(num, reverse=True))
+  return int(ret) if isinstance((ret:=-(num//-amt)), float) else ret
```

Comment:
Not sure if there's a less hacky way to check `UOp`, maybe only moving to a separate file